### ISSUE TITLE
Storing and retrieving user account data

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#14643]
_Originally posted by @baiti in https://github.com/cgeo/cgeo/issues/14634#issuecomment-1729476916_

Slightly OT ... but still somewhat related: I spent some time looking at the code and it seems we don't have a concept of managing a `userid` based on the connector. `Settings.getUserName()` seems to give me the GC user. What if somebody wants to exlusively use OC and not GC? What if someone maintains different usernames on GC and OC?

I think this is a bigger problem than just the "visitedDate" and further, I do think it should be resolved one way or another.

In the OC case: there is an OKAPI service called "userinfo". This returns the username, among other things. But it doesn't make any sense to call that service each and every time I want to parse a geocache's log for the `visitedDate`. It must be possible, to store that variable somewhere I can later access it without running a costly OKAPI method.

I have tried to find such a way but my Java skills are way way way not sufficient to accomplish such a task. I repeatedly ran into  compile errors because things were "static" where they shouldn't be static and the othe way around. The error messages I got just confused me more, the more I tried.

**Comment by moving-bits** [comment]
Some more info on this:
- `getUserName()` is solely used for GC connector and retrieves the value stored in `username`.
- we do have `getCredentials()` and `setCredentials()` methods used by those connectors that are using a login/password scheme
- OKAPI-based connectors use a OAuth mechanism. No login/password is stored locally, only a temporary access token.

So an approach could be:
- Use the mentioned "userinfo" service to retrieve the userinfo for OAuth-based connectors and store that info using `setCredential()` (or a new, similar working connector-based method storing and retrieving only a username). 
- This method should be called on each authorization to an OAuth-based service (because user may have changed their login data) as well as on connecting to an OAuth-based service if no username for this connector is stored already (for migration).

**Comment by moving-bits** [comment]
Additional finding: Main screen connector info seems to be able to distinguish usernames per connector. Have a look at `FoundNumCounter.getNotBlankUserName()` call in `MainActivity.java` => `UpdateUserInfoHandler` => `fillView`.

**Comment by fm-sys** [comment]
Yes, we already store the user name for the FoundNumCounter on a per-connector base...
